src/plugins/intel_gna/src/transformations/remove_pre_post_processing.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/pass/graph_rewrite.hpp>
+
+#include "gna_data_types.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+
+/**
+ * @brief Remove Transpose/Gater layers connected to Inputs and create preprocessing model
+ * to support input preprocessing on CPU
+ *
+ * Searches for next pattern
+ *     Any input layer
+ *           |
+ *    Transpose/Gather
+ *           |
+ *        Any layer
+ *
+ * And transforms to
+ *     Any input layer
+ *           |
+ *        Any layer
+ */
+class RemoveInputsProcessing : public ov::pass::ModelPass {
+public:
+    OPENVINO_RTTI("RemoveInputsProcessing", "0");
+    RemoveInputsProcessing(ov::intel_gna::PrePostProcessModels* subgraph_cpu_map = nullptr)
```

Comment:
Just wonder if we could pass here PrePostProcess class containing subgraph_cpu_map  instead of passing raw pointer to ov::intel_gna::PrePostProcessModels*
class could contain method 
```update_subgraph(std::string name, std::shared_ptr<ov::Model> model)
{
// emplace
}
```

I think this would do implementation loosely tied - currently to stick to internal implementation details.

Reply:
what is the benefit to use class here if we collect models into the map? Not clear what do you suggest